src/gui/qgsfeaturelistcombobox.cpp

```diff
@@ -203,6 +203,14 @@ void QgsFeatureListComboBox::keyPressEvent( QKeyEvent *event )
   QComboBox::keyReleaseEvent( event );
 }
 
+bool QgsFeatureListComboBox::event( QEvent *event )
+{
+  if ( event->type() == QEvent::EnabledChange )
+    mLineEdit->setShowClearButton( true );
```

Comment:
I admit I don't follow this: shouldn't you check if the widget is enabled and setShowClearButton accordingly?

Reply:
in any event, make sure that this does not show the clear button when it's not needed (value is NULL).

Reply:
As far I understood the `setShowClearButton( true )` it only sets that the clear button is required on the usual circumstances (not null, enabled, not read-only) handled here https://github.com/qgis/QGIS/blob/master/src/gui/qgsfilterlineedit.cpp#L183

It could call `updateClearIcon` here as well but then I need to change it to public.

Reply:
what about implementing this logic directly inside the lineedit? Wouldn't be more logical? Or is there a reason why the `EnabledChange` event does not apply to the lineedit.